src/server/race/RaceDirector.cpp

```diff
@@ -1331,6 +1335,82 @@ void RaceDirector::HandleStartRace(
   if (clientContext.characterUid != raceInstance.masterUid)
     throw std::runtime_error("Client tried to start the race even though they're not the master");
 
+  // Check if all requirements are met to start the race
+  bool allPlayersReady = false;
+  bool isTeamMode = false;
+  uint32_t redTeamCount = 0;
+  uint32_t blueTeamCount = 0;
+
+  _serverInstance.GetRoomSystem().GetRoom(
+    clientContext.roomUid,
+    [&allPlayersReady, &isTeamMode, &redTeamCount, &blueTeamCount, clientContext](Room& room)
+    {
+      // Toggle the ready state of the master
+      room.GetPlayer(clientContext.characterUid).ToggleReady();
+
+      allPlayersReady = std::ranges::all_of(
+        room.GetPlayers() | std::views::values,
+        [](const Room::Player& player)
+        {
+          return player.IsReady();
+        });
+      
+      isTeamMode = room.GetRoomDetails().teamMode == Room::TeamMode::Team;
+
+      // Count team members if in team mode
+      if (isTeamMode)
+      {
+        for (const auto& player : room.GetPlayers() | std::views::values)
+        {
+          switch (player.GetTeam())
+          {
+            case Room::Player::Team::Red:
+              redTeamCount++;
+              break;
+            case Room::Player::Team::Blue:
+              blueTeamCount++;
+              break;
+            default:
+              break;
```

Comment:
```suggestion
            case Room::Player::Team::Red: 
            {
              redTeamCount++;
              break;
            }
            
            case Room::Player::Team::Blue: 
            {
              blueTeamCount++;
              break;
            }
            
            default: 
            {
              break;
            }
```
I think we should do it this way. It's much more cleaner.